Applications/Spire/Source/Ui/DropShadow.cpp

```diff
@@ -44,7 +44,7 @@ DropShadow::DropShadow(bool is_menu_shadow, bool has_top, QWidget* parent)
   setAttribute(Qt::WA_ShowWithoutActivating);
   m_parent->window()->installEventFilter(this);
   connect(m_parent, qOverload<QObject*>(&QObject::destroyed),
-    [=] (auto object) { deleteLater(); });
+    this, &DropShadow::on_parent_destroyed);
```

Comment:
Why do you need a method now?

Reply:
I was having issues with what looked like the lambda getting prematurely destroyed.

Reply:
A lambda, as a syntactic construct, cannot be destroyed per se. The closure it is assigned to can, but I think `connect` takes functors by copy so it should not be the case (and otherwise you would get some warnings). I think the only difference is that you pass the context `this` in the newer version, but you can do so with a lambda as well. Have you tried
```
  connect(m_parent, qOverload<QObject*>(&QObject::destroyed),
    [=] (auto object) { deleteLater(); });
    this, [=] (auto object) { deleteLater(); });
```
?

Reply:
This was actually related to an issue I fixed in the key bindings branch. I'll revert it back to using the lambda.

Reply:
I missed your comment before I replied, but the issue of semantics is something I should keep in mind.